components/common/src/command/package/install.rs

```diff
@@ -815,8 +848,14 @@ impl<'a> InstallTask<'a> {
     ) -> Result<()> {
         // Canonicalize both paths to ensure that there aren't any symlinks when comparing them
         // later.
+
         let artifact_path = artifact_path.canonicalize()?;
-        let cache_path = self.cached_artifact_path(ident).canonicalize()?;
+        let mut cache_path = self.cached_artifact_path(ident);
+        // This call can fail to canonicalize paths under certain circumstances. This will allow
```

Comment:
Sure, specifically I found failures here working from within the Linux Subsystem on Windows. It's a deep syscall failure that we don't really control. I inspected this code and made the change because the intention of this code is to _try our best_ which should never be a code path which fails out